docs/markdown/Build-options.md

```diff
@@ -262,7 +262,7 @@ visual studio. It should be the same name as an executable target
 name.
 
 ```meson
-project('my_project', 'c', default_options: ['backend_startup_project=my_exe'])
+project('my_project', host_machine_languages : 'c', default_options: ['backend_startup_project=my_exe'])
```

Comment:
I really do not like this. The workflow should instead be "try to enable C for both build and host, if either one is not available just continue and if the build definition then tries to use a language that does not exist, error out with an informal message". That is what 99% of project want 99% of the time so make it the default and not require people to write boilerplate in their build files.

Reply:
I think the current meaning of 'c' is fine, but given the current patch I would disallow mixing the positional version with `*_languages`.

Using a dictionary has the advantage that it automatically forbids such mixed use, because it doesn't add any new keyword argument.

I agree that we should at least add `machine: str` as a replacement for `native: bool` in the same release, if we go that way.

Reply:
That argument for using a dictionary is pretty compelling to me.